pyccel/codegen/printing/ccode.py

```diff
@@ -2810,6 +2819,26 @@ def _print_SetDiscard(self, expr):
         arg_val = self._print(expr.args[0])
         return f'{var_type}_erase({set_var}, {arg_val});\n'
 
+    #================== Dict methods ==================
+
+    def _print_DictPop(self, expr):
+        target = expr.dict_obj
+        class_type = target.class_type
+        c_type = self.get_c_type(class_type)
+
+        dict_var = self._print(ObjectAddress(target))
+        key = self._print(expr.key)
+        tmp_var = self.scope.get_temporary_variable(class_type.value_type)
+
+        if expr.default_value:
+            default = self._print(expr.default_value)
+            pop_expr = f"{tmp_var.name} = {c_type}_pop_with_default({dict_var}, {key}, {default});"
+        else:
+            pop_expr = f"{tmp_var.name} = {c_type}_pop({dict_var}, {key});"
+        self._additional_code += f"{pop_expr}\n"
+
+        return tmp_var.name
```

Comment:
Why do you need to create a temporary variable to hold the result of `pop`? Can't you just return the function call?